modules/reindex/src/main/java/org/elasticsearch/reindex/AbstractAsyncBulkByScrollAction.java

```diff
@@ -478,6 +510,12 @@ void onBulkResponse(BulkResponse response, Runnable onSuccess) {
                 return;
             }
 
+            if (scrollSource.hasScroll() == false) {
```

Comment:
I think we should turn this into an assert instead, asserting that we never get here if there is no scroll source? Perhaps I missed a case where we would get here?

Reply:
This could be reached when `max_docs` is set, but the source index has less docs.

Reply:
Related to occasional failures of `org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT › test {yaml=reference/docs/reindex/line_876}` in this branch

Reply:
I see, thanks. It is a slightly convoluted way to find out that we exhausted the search. I do see that it requires a bit more change (or at least examination) to find out that the search is exhausted. We can let this be like it is for now, but perhaps add a comment about this case being about the search being exhausted?